com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/PathTracing.cs

```diff
@@ -306,26 +322,52 @@ class RenderPathTracingData
         {
             public PathTracingParameters parameters;
             public TextureHandle output;
+            public TextureHandle sky;
         }
 
-        TextureHandle RenderPathTracing(RenderGraph renderGraph, in PathTracingParameters parameters, TextureHandle pathTracingBuffer)
+        TextureHandle RenderPathTracing(RenderGraph renderGraph, in PathTracingParameters parameters, TextureHandle pathTracingBuffer, TextureHandle skyBuffer)
         {
             using (var builder = renderGraph.AddRenderPass<RenderPathTracingData>("Render PathTracing", out var passData))
             {
                 passData.parameters = parameters;
                 passData.output = builder.WriteTexture(pathTracingBuffer);
+                passData.sky = builder.ReadTexture(skyBuffer);
 
                 builder.SetRenderFunc(
                     (RenderPathTracingData data, RenderGraphContext ctx) =>
                     {
-                        RenderPathTracing(data.parameters, data.output, ctx.cmd);
+                        RenderPathTracing(data.parameters, data.output, data.sky, ctx.cmd);
                     });
 
                 return passData.output;
             }
         }
 
-        TextureHandle RenderPathTracing(RenderGraph renderGraph, HDCamera hdCamera)
+        // Simpler variant used by path tracing, without depth buffer or volumetric computations
+        void RenderSky(RenderGraph renderGraph, HDCamera hdCamera, TextureHandle skyBuffer)
+        {
+            if (m_CurrentDebugDisplaySettings.DebugHideSky(hdCamera))
+                return;
+
+            using (var builder = renderGraph.AddRenderPass<RenderSkyPassData>("Render Sky for Path Tracing", out var passData))
+            {
+                passData.visualEnvironment = hdCamera.volumeStack.GetComponent<VisualEnvironment>();
+                passData.sunLight = GetCurrentSunLight();
+                passData.hdCamera = hdCamera;
+                passData.colorBuffer = builder.WriteTexture(skyBuffer);
+                passData.depthTexture = builder.WriteTexture(CreateDepthBuffer(renderGraph, true, false));
+                passData.debugDisplaySettings = m_CurrentDebugDisplaySettings;
+                passData.skyManager = m_SkyManager;
+
+                builder.SetRenderFunc(
+                    (RenderSkyPassData data, RenderGraphContext context) =>
+                    {
+                        data.skyManager.RenderSky(data.hdCamera, data.sunLight, data.colorBuffer, passData.depthTexture, data.debugDisplaySettings, context.cmd);
```

Comment:
passdata.depthTexture should be data.depthTexture, otherwise we will get GCAlloc during rendering as far as I know.
(we had similar issues in many places in the past)

Reply:
Yes, this will generate GCAlloc. Simple way to check that you don't capture any variables in the lambda is to hover the mouse on the  => in visual studio. It will tell you if anything is captured.

Reply:
Oh yeah that was a typo, thanks! Fixed now.